dalm/pipelines/reading_comprehension_pipeline.py

```diff
@@ -0,0 +1,383 @@
+            generation_state["processed_texts"].append(text_identifier)
+            generation_state["total_texts"] += 1
+            pickle.dump(generation_state, open(generation_state_file, "wb"))
+
+        logger.info(" Statistics ")
+        success_files_count = generation_state["total_texts"] - generation_state["texts_missed"]
+        logger.info(f"Total number of successfully extracted q&a: {success_files_count}")
+        logger.info(f"Total texts missed: {generation_state['texts_missed']} out of {generation_state['total_texts']}")
+        logger.info(f"Total texts processed: {generation_state['total_texts']}")
+
+        for file in os.listdir(llm_kwargs.dataset_output_path):
+            with open(os.path.join(llm_kwargs.dataset_output_path, file), "r") as f:
+                in_memory_dataset.append({"messages": json.load(f)})
+
+    if in_memory_dataset == []:
+        raise ValueError("No dataset generated")
+
+    # shuffle in memory dataset
+    random.shuffle(in_memory_dataset)
+
+    dataset = datasets.Dataset.from_list(in_memory_dataset)
+
+    dataset.save_to_disk(output_dataset_name)
+
+    train_generator(
+        model_name=model_name,
+        dataset_name=output_dataset_name,
+        num_train_epochs=num_train_epochs,
+        split=split,
+        size_valid_set=size_valid_set,
+        streaming=streaming,
+        shuffle_buffer=shuffle_buffer,
+        seq_length=seq_length,
+        num_workers=num_workers,
+        eval_steps=eval_steps,
+        logging_steps=logging_steps,
+        per_device_train_batch_size=per_device_train_batch_size,
+        per_device_eval_batch_size=per_device_eval_batch_size,
+        gradient_accumulation_steps=gradient_accumulation_steps,
+        gradient_checkpointing=gradient_checkpointing,
+        group_by_length=group_by_length,
+        packing=packing,
+        lora_alpha=lora_alpha,
+        lora_dropout=lora_dropout,
+        lora_r=lora_r,
+        learning_rate=learning_rate,
+        lr_scheduler_type=lr_scheduler_type,
+        num_warmup_steps=num_warmup_steps,
+        weight_decay=weight_decay,
+        optimizer_type=optimizer_type,
+        output_dir=model_output_dir,
+        neftune_noise_alpha=neftune_noise_alpha,
+        log_with=log_with,
+        local_dataset=True,
+        validation_split=validation_split,
+        run_name=run_name,
+    )
+
+    if generation_state_file:
+        os.remove(generation_state_file)
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_name", type=str, default="HuggingFaceH4/zephyr-7b-alpha", help="name of the model to be trained"
+    )
+    parser.add_argument(
+        "--output_dataset_name",
+        type=str,
+        required=True,
+        help="name of the dataset that the generated data will be saved to",
+    )
+    parser.add_argument(
+        "--comprehension_type",
+        type=SynthMode,
+        default=SynthMode.BOTH,
```

Comment:
```suggestion
        default=SynthMode.LLM,
```

I think we should default to `SynthMode.LLM` since it has undergone more testing and so it is more likely to work, and requires less parameters.  Also it appears to work on more datasets than the regex approach.

Users are likely to use the default, so `SynthMode.LLM` will probably provide a better first time experience than `SynthMode.BOTH`.  They can always set it to use regex or both if they feel adventurous.

cc @shamanez